### PR TITLE
Chore/add perms check in projects api query

### PR DESCRIPTION
**Pull request by joshenlim**
- Add perms check in project-api-query to prevent continuous calling if user is read only mode and endpoint will not return any API keys
- Start deprecating use of project-api-query, shift to newly added projects-settings-v2-query  for the following parts of the dashboard:
    - API Docs
    - Graphiql
    - S3 Connection
    - Command Menu
    - Connect UI
    - Auth/users
        - Create user modal
        - Ban user modal
    - Home page API keys section
    - StorageLayout: Initialise storage store
- We need to deprecate project-api-query and project-settings-query so I'll likely do the refactor in multiple parts as there's many changes involved, makes it easier to test stuff too

### CODE REVIEW

File: apps/studio/components/interfaces/App/CommandMenu/ApiKeys.tsx [hunk 1]
```diff
@@ -23,14 +23,16 @@ export function useApiKeysCommands() {
...
     { enabled: !!project }
   )
   const ref = project?.ref || '_'
 
-  const anonKey = settings?.autoApiService?.defaultApiKey ?? undefined
-  const serviceKey = settings?.autoApiService?.serviceApiKey ?? undefined
+  const anonKey = (settings?.service_api_keys ?? []).find((x) => x.tags === 'anon')?.api_key
+  const serviceKey = (settings?.service_api_keys ?? []).find(
+    (x) => x.tags === 'service_role'
+  )?.api_key
```
Comment (alaister) [inline apps/studio/components/interfaces/App/CommandMenu/ApiKeys.tsx:35]: I'm seeing this a lot in this PR... worth a cheeky function? Something like:
```typescript
function getDefaultAPIKeys(settings?: ProjectSettings) {
  const anonKey = (settings?.service_api_keys ?? []).find((x) => x.tags === 'anon')?.api_key
  const serviceKey = (settings?.service_api_keys ?? []).find(
    (x) => x.tags === 'service_role'
  )?.api_key

  return { anonKey, serviceKey }
}
```

wdyt?

Comment (joshenlim) [reply apps/studio/components/interfaces/App/CommandMenu/ApiKeys.tsx]: okay just pushed a fix! 🙏

**Review by alaister — APPROVED** [review]
LGTM!

- Tested everything mentioned in the PR description on preview

Also a few nits:

File: apps/studio/data/config/project-api-query.ts [hunk 2]
```diff
@@ -32,24 +34,25 @@ export async function getProjectApi({ projectRef }: ProjectApiVariables, signal?
 export type ProjectApiData = Awaited<ReturnType<typeof getProjectApi>>
 export type ProjectApiError = ResponseError
 
+/** @deprecated */
```
[the remark this answers is not in the document]
Comment (joshenlim) [reply apps/studio/data/config/project-api-query.ts:37]: i'll leave this for now - cause the next PRs will delete this file entirely haha

File: apps/studio/data/config/project-settings-query.ts [hunk 3]
```diff
@@ -42,6 +42,7 @@ export async function getProjectSettings(
 export type ProjectSettingsData = Awaited<ReturnType<typeof getProjectSettings>>
 export type ProjectSettingsError = ResponseError
 
+/** @deprecated */
```
[the remark this answers is not in the document]
Comment (joshenlim) [reply apps/studio/data/config/project-settings-query.ts:45]: likewise will leave this for now

**Pull request merged**